test/com/google/javascript/jscomp/PeepholeRemoveDeadCodeTest.java

```diff
@@ -732,6 +740,17 @@ public void testFoldAssign() {
     testSame("x.a=x.a");
     test("var y=(x=x)", "var y=x");
     test("y=1 + (x=x)", "y=1 + x");
+
+    // GitHub issue #1745: https://github.com/google/closure-compiler/issues/1745
+    test("Object.create({}).prop = 1", "");
+    test("Object.create(proto).foo.bar.baz = 1", "");
+
+    // add() is side-effect free, but addClass() is not
+    test("var jQuery = new $; jQuery.add().add().bar = 1", "var jQuery = new $;");
+    test("var jQuery = new $; jQuery.add().bar.add().baz = 1", "var jQuery = new $;");
+    test("var jQuery = new $; jQuery.bar.add().baz = 1", "var jQuery = new $;");
```

Comment:
I think you are right. I'm not sure how to determine what the "leftmost receiver" is? For example, in:
```js
Object.create({}).prop = 1;
```
Should `Object` be considered a pre-existing object? We'd like to remove the assignment though.